application/src/main/java/com/sanction/thunder/ThunderApplication.java

```diff
@@ -29,9 +33,34 @@ public void run(ThunderConfiguration config, Environment env) {
         .thunderModule(new ThunderModule(env.metrics()))
         .build();
 
-    // Authentication
-    env.jersey().register(AuthFactory.binder(new BasicAuthFactory<>(new ThunderAuthenticator(
-        config.getApprovedKeys()), "THUNDER - AUTHENTICATION", Key.class)));
+    // auth filter one
+//    AuthFilter userCredentialAuthFilter = new BasicCredentialAuthFilter.Builder<User>()
+//        .setAuthenticator(component.getUserAuthenticator())
+//        .setRealm("THUNDER - USER - AUTHENTICATION")
+//        .buildAuthFilter();
+//
+//    // auth filter one
+//    AuthFilter keyCredentialAuthFilter = new BasicCredentialAuthFilter.Builder<Key>()
+//        .setAuthenticator(new KeyAuthenticator(config.getApprovedKeys()))
+//        .setRealm("THUNDER - KEY - AUTHENTICATION")
+//        .buildAuthFilter();
+    env.jersey().register(new AuthDynamicFeature(new BasicCredentialAuthFilter.Builder<User>()
+        .setAuthenticator(component.getUserAuthenticator())
+        .setRealm("THUNDER - AUTHENTICATION")
+        .buildAuthFilter()));
+
+    env.jersey().register(new AuthDynamicFeature(new BasicCredentialAuthFilter.Builder<Key>()
+        .setAuthenticator(new KeyAuthenticator(config.getApprovedKeys()))
```

Comment:
Can we make this one the same as `UserAuthenticator` and get it from `ThunderComponent` (use Dagger)?